docs/examples/builds/multi-root-editor.md

```diff
@@ -0,0 +1,187 @@
+toc: false
+classes: main__content--no-toc
+---
+
+# Multi-root editor
+
+Multi-root editor is an editor type that features multiple, separate editable areas.
+
+The main difference between a multi-root editor and using multiple separate editors (like in the {@link examples/builds/inline-editor inline editor demo}) is the fact that in a multi-root editor all editable areas belong to the same editor instance and produce one document, share the same configuration, toolbar and the undo stack.
+
+{@snippet examples/multi-root-editor}
+
+## Editor example configuration
+
+Check out the {@link installation/getting-started/predefined-builds#installation-example Quick start} guide to learn more about implementing this kind of editor. You will find implementation steps there. You can see this example editor’s code below.
+
+<details>
+<summary>View editor configuration script</summary>
+
+```js
+import MultiRootEditor from '@ckeditor/ckeditor5-build-multi-root/src/ckeditor';
+
+MultiRootEditor
+	.create(
+		// Define roots / editable areas:
+		{
+			header: document.querySelector( '#header' ),
+			content: document.querySelector( '#content' ),
+			leftSide: document.querySelector( '#left-side' ),
+			rightSide: document.querySelector( '#right-side' )
+		},
+		// Editor configration:
+		{
+			cloudServices: {
+				// All predefined builds include the Easy Image feature.
+				// Provide correct configuration values to use it.
+				tokenUrl: 'https://example.com/cs-token-endpoint',
+				uploadUrl: 'https://your-organization-id.cke-cs.com/easyimage/upload/'
+				// Read more about Easy Image - https://ckeditor.com/docs/ckeditor5/latest/features/images/image-upload/easy-image.html.
+				// For other image upload methods see the guide - https://ckeditor.com/docs/ckeditor5/latest/features/images/image-upload/image-upload.html.
+			}
+		}
+	)
+	.then( editor => {
+		window.editor = editor;
+
+		// Append toolbar to a proper container.
+		const toolbarContainer = document.querySelector( '#toolbar' );
+		toolbarContainer.appendChild( editor.ui.view.toolbar.element );
+
+		// Make toolbar sticky when the editor is focused.
+		editor.ui.focusTracker.on( 'change:isFocused', () => {
+			if ( editor.ui.focusTracker.isFocused ) {
+				toolbarContainer.classList.add( 'sticky' );
+			} else {
+				toolbarContainer.classList.remove( 'sticky' );
+			}
+		} );
+	} )
+	.catch( error => {
+		console.error( 'There was a problem initializing the editor.', error );
+	} );
+
+```
+
+</details>
+
+<details>
+<summary>View editor content listing</summary>
+
+```html
+<div id="toolbar"></div>
+<!--
+	Wrapping the structure inside a pair of
+	contenteditable="true" + contenteditable="false" elements
+	is required to provide proper caret handling when
+	using arrow keys at the start and end of an editable area.
+	
+	You can skip them if you don't want to move the
+	caret between editable areas using arrow keys.
+!--> 
+<div contenteditable="true">
+	<div contenteditable="false">
```

Comment:
This doesn't look good but it is necessary to have proper arrows control. But maybe you know better solutions? By adding this, we don't have to introduce logic that would be fired when arrow key is pressed at the beginning or at the end of an editable. The browser handles it by itself.

However, it may add some quirkiness...? OTOH, we use similar structures this inside the editor (e.g. tables).

Reply:
cc @Reinmar